lab2/app.js

```diff
@@ -0,0 +1,41 @@
+
+// PubSub is a single object for publishing data to multiple subscribers.
+class PubSub {
+	constructor () {
+		this.subscribers = [];
+	}
+
+	// Adds a new function to the list of existing functions.
+	subscribe (fn) {
+		this.subscribers.push(fn);
+	}
+
+	// Calls the functions that the object, "data", is subscribed to.
+	publish (data) {
+		this.subscribers.forEach(subscriber => {
+			subscriber(data);
+		});
+	}
+}
+
+// Create a PubSub object
+const pubSub = new PubSub();
+
+
+let stawberry_count = document.getElementById("strawberry_count");
+
+// Define an increment function
+function increment () {
+	incrementalGame.state.counter++;
+	strawberry_count.innerHTML = "Strawberries " + incrementalGame.state.counter;
```

Comment:
Maybe it's better to add a span after "Strawberries" so that you can just target the span and add the counter there instead of having printing Strawberries all the time here.

Reply:
Thanks for the tip Eric! I'll make the changes as soon as possible.